_tests/test_endpoints.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+import requests
+import pytest
+from collections import namedtuple
+import glob
+import os
+import json
+import logging
+
+# Setup basic configuration for logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+EndpointTest = namedtuple('EndpointTest', ['chain', 'endpoint', 'provider', 'address'])
+
+# Set this to False to ignore the whitelist and process all providers
+use_whitelist = True
+
+# Whitelist for specific chains and providers
+whitelist = {
+    "chains": [
+        "axelar",
+        "celestia",
+        "composable",
+        "cosmoshub",
+        "dydx",
+        "dymension",
+        "evmos",
+        "injective",
+        "neutron",
+        "noble",
+        "osmosis",
+        "stargaze",
+        "stride"
```

Comment:
If we want to ensure that our CI passes only when all providers are processed, we could also keep a tally of the number of successful chains validated, checking that it matches the length of the whitelisted chains at the end.